libs/common/src/sql-db-entities/payment/payment.entity.test.ts

```diff
@@ -0,0 +1,50 @@
+import { FeeRecordEntityMockBuilder, PaymentEntityMockBuilder, UtilisationReportEntityMockBuilder } from '../../test-helpers';
+
+describe('PaymentEntity', () => {
+  const utilisationReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build();
+
+  describe('updateWithAdditionalFeeRecords', () => {
+    it('adds new fee records to the payment', () => {
+      // Arrange
+      const payment = PaymentEntityMockBuilder.forCurrency('GBP').build();
+
+      const existingFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).build();
+      payment.feeRecords = [existingFeeRecord];
+
+      const newFeeRecord1 = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(2).build();
+      const newFeeRecord2 = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(3).build();
+
+      const userId = 'abc123';
+
+      // Act
+      payment.updateWithAdditionalFeeRecords({
+        feeRecords: [newFeeRecord1, newFeeRecord2],
+        requestSource: { platform: 'TFM', userId },
+      });
+
+      // Assert
+      expect(payment.feeRecords).toEqual(expect.arrayContaining([existingFeeRecord, newFeeRecord1, newFeeRecord2]));
+    });
+
+    it('does not allow duplicate fee records to be added to the payment', () => {
+      // Arrange
+      const payment = PaymentEntityMockBuilder.forCurrency('GBP').build();
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).build();
+      payment.feeRecords = [feeRecord];
+
+      const userId = 'abc123';
+
+      // Act & Assert
+      expect(() => {
+        payment.updateWithAdditionalFeeRecords({
+          feeRecords: [feeRecord],
```

Comment:
Please can we use a separate in memory object for the existing one and the new one, our code doesn't and shouldn't rely on them being the same in memory so good to avoid in the test too I think.

Reply:
This test has been removed (other comment) but yeah I agree it'd be better like the other test case

Reply:
Reintroduced test (other comment), made the mock a separate object with the same ID and improved naming of these mocks.